src/matchPath.js

```diff
@@ -47,7 +47,7 @@ function matchPathBase(end, routePath, urlPath) {
   const params = Object.create(null);
   const path = m[0];
 
-  for (let i = 1; i < m.length; i++) {
+  for (let i = 1; i < m.length; i += 1) {
```

Comment:
From Airbnb JavaScript Style Guide:

- [13.6](https://github.com/airbnb/javascript#variables--unary-increment-decrement) Avoid using unary increments and decrements (++, --). eslint [`no-plusplus`](http://eslint.org/docs/rules/no-plusplus)

  > Why? Per the eslint documentation, unary increment and decrement statements are subject to automatic semicolon insertion and can cause silent errors with incrementing or decrementing values within an application. It is also more expressive to mutate your values with statements like `num += 1` instead of `num ++`. Disallowing unary increment and decrement statements also prevents you from pre-incrementing/pre-decrementing values unintentionally which can also cause unexpected behavior in your programs.